safe_transaction_service/contracts/tx_decoder.py

```diff
@@ -552,6 +552,34 @@ class DbTxDecoder(TxDecoder):
 
     cache_abis_by_address = TTLCache(maxsize=2048, ttl=60 * 5)  # 5 minutes of caching
 
+    @classmethod
+    def get_most_relevant_contract_abi(cls, address: ChecksumAddress):
+        """
+        Retrieves the most relevant ABI for the contract at the given address.
+
+        :param address: Contract address
+        :return: List of ABI data if found, `None` otherwise.
+        """
+        return (
+            ContractAbi.objects.filter(contracts__address=address)
+            .order_by("relevance")
+            .values_list("abi", flat=True)
+            .first()
+        )
+
+    def contract_has_fallback_call(self, address: ChecksumAddress) -> bool:
```

Comment:
This method is not using the ABI cache

Reply:
I added a new chache for the abis by address and kept the cache with the selectors for the abi functions by address.